client_sdk/avalon_client_sdk/fabric/chaincode/receipt/go/main.go

```diff
@@ -0,0 +1,343 @@
+//   bytes32[] appTypeIds
+// returns:
+func (t *Receipt) workOrderReceiptCreate(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	logger.Info("workOrderReceiptCreate")
+	if len(args) != 6 {
+		logger.Errorf("Mismatch number of arguments, expect 6, received %d", len(args))
+		return shim.Error("workOrderReceiptCreate must include 6 arguments, workOrderId, workerId, workerServiceId, requesterId, receiptCreateStatus and workOrderRequestHash")
+	}
+
+	arg4, err := strconv.ParseUint(args[4], 10, 64)
+	if err != nil {
+		logger.Errorf("Receipt create status must be an integer")
+		return shim.Error("Receipt create status must be an integer")
+	}
+
+	var r Receipt
+	r.WorkOrderId = args[0]
+	r.WorkerId = args[1]
+	r.WorkerServiceId = args[2]
+	r.RequesterId = args[3]
+	r.ReceiptCreateStatus = arg4
+	r.WorkOrderRequestHash = args[5]
+
+	//Serialize the value
+	value, err := json.Marshal(r)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	err = stub.PutState(r.WorkOrderId, value)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	compValue := []byte(r.WorkOrderId)
+
+	attrs, _ := processAttributes([]string{r.WorkerServiceId, r.WorkerId, r.RequesterId, args[4]},
+		[]string{BYTE32FORMAT, BYTE32FORMAT, BYTE32FORMAT, UINT64FORMAT})
+	compKey, err := stub.CreateCompositeKey(OBJECTTYPE, attrs)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+	err = stub.PutState(compKey, compValue)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	// Handling payload for the event
+	var eventData ReceiptCreatedEvent
+	eventData.WorkOrderId = r.WorkOrderId
+	eventData.WorkerServiceId = r.WorkerServiceId
+	eventData.WorkerId = r.WorkerId
+	eventData.RequesterId = r.RequesterId
+	eventData.ReceiptStatus = r.ReceiptCreateStatus
+	eventData.WorkOrderRequestHash = r.WorkOrderRequestHash
+	eventData.ErrorCode = 0
+
+	eventPayload, err := json.Marshal(eventData)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	err = stub.SetEvent("workOrderReceiptCreated", eventPayload)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	logger.Info("Finished workOrderReceiptCreate")
+	return shim.Success(nil)
+}
+
+// workOrderReceiptUpdate - This function updates a Receipt
+// params:
+//   bytes32 workOrderId,
+//   bytes32 updaterId. should this be workerId?
+//   uint256 updateType,
+//   bytes updateData,
+//   bytes updateSignature,
+//   string signatureRules
+func (t *Receipt) workOrderReceiptUpdate(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	logger.Info("workOrderReceiptUpdate")
+
+	return shim.Error("Not Implemented")
+}
+
+// workOrderReceiptLookUp - This function retrieves a list of registry ids that match input
+// parameter. The registry must match to all input parameters (AND mode) to be
+// included in the list.
+// params:
+//    bytes32 workerServiceId,
+//    bytes32 workerId,
+//    bytes32 requesterId,
+//    uint256 receiptStatus
+// returns:
+//   int totalCount
+//   string LookupTag
```

Comment:
Change `LookupTag` to `lookupTag`
(see line 46)

Reply:
ok